lib/plugins/bull-mq-metrics/queueDiscoverers.ts

```diff
@@ -2,17 +2,28 @@ import { backgroundJobProcessorGetActiveQueueIds } from '@lokalise/background-jo
 import type { Redis } from 'ioredis'
 
 export type QueueDiscoverer = {
-  discoverQueues: () => Promise<string[]>
+  discoverQueues: () => Promise<RedisQueue[]>
+}
+
+type RedisQueue = {
+  redisInstance: Redis
+  queueName: string
 }
 
 export class RedisBasedQueueDiscoverer implements QueueDiscoverer {
   constructor(
-    private readonly redis: Redis,
+    private readonly redisInstances: Redis[],
     private readonly queuesPrefix: string,
   ) {}
 
-  async discoverQueues(): Promise<string[]> {
-    const scanStream = this.redis.scanStream({
+  async discoverQueues(): Promise<RedisQueue[]> {
+    return Promise.all(
```

Comment:
there can potentially be a lot of them (e. g. AP has ~10, I think?), I wonder if promise-pool would make sense here

Reply:
Good point, It should be one-time operation, but still makes sense to add some concurrency limitation. Added PromisePool and extracted that part to abstract class to have common approach